### PR TITLE
engine(externalresource): add cleanPersistResource method for filemananger

### LINKED ISSUE #7485: dmctl doesn't support platform darwin/amd64

### LINKED ISSUE #7486: panic: inconsistent executor ID of s3 file
### What did you do?

chaos test

### What did you expect to see?

_No response_

### What did you see instead?
```
[2022/10/27 07:39:34.470 +00:00] [INFO] [worker_manager.go:354] ["start to create worker"] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [component=worker_manager] [task_id=replica-02] [unit=DMDumpTask]
[2022/10/27 07:39:34.470 +00:00] [INFO] [master.go:707] [CreateWorker] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [worker-type=DMDumpTask] [worker-config="{\"task-mode\":\"all\",\"shard-mode\":\"\",\"ignore-checking-items\":[],\"timezone\":\"\",\"collation_compatible\":\"loose\",\"target-database\":{\"host\":\"tidb-0.tidb\",\"port\":4000,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":{\"tidb_txn_mode\":\"optimistic\"},\"security\":null},\"shadow-table-rules\":[],\"trash-table-rules\":[],\"filters\":{},\"expression-filter\":{},\"block-allow-list\":{\"balist-01\":{\"do-tables\":[],\"do-dbs\":[\"dmjob\"],\"ignore-tables\":[],\"ignore-dbs\":[]}},\"mydumpers\":{},\"loaders\":{},\"syncers\":{},\"routes\":{},\"validators\":{},\"upstreams\":[{\"SourceID\":\"replica-02\",\"Meta\":null,\"FilterRules\":null,\"ColumnMappingRules\":null,\"RouteRules\":null,\"ExpressionFilters\":null,\"BWListName\":\"\",\"BAListName\":\"balist-01\",\"MydumperConfigName\":\"\",\"Mydumper\":{\"mydumper-path\":\"./bin/mydumper\",\"threads\":4,\"chunk-filesize\":\"64\",\"statement-size\":0,\"rows\":0,\"where\":\"\",\"skip-tz-utc\":true,\"extra-args\":\"\"},\"MydumperThread\":0,\"LoaderConfigName\":\"\",\"Loader\":{\"pool-size\":16,\"dir\":\"./dumped_data\",\"import-mode\":\"sql\",\"on-duplicate\":\"replace\"},\"LoaderThread\":0,\"SyncerConfigName\":\"\",\"Syncer\":{\"meta-file\":\"\",\"worker-count\":16,\"batch\":100,\"queue-size\":1024,\"checkpoint-flush-interval\":30,\"compact\":false,\"multiple-rows\":false,\"max-retry\":0,\"auto-fix-gtid\":false,\"enable-gtid\":false,\"disable-detect\":false,\"safe-mode\":false,\"safe-mode-duration\":\"60s\",\"enable-ansi-quotes\":false},\"SyncerThread\":0,\"ContinuousValidatorConfigName\":\"\",\"ContinuousValidator\":{\"mode\":\"none\",\"worker-count\":0,\"validate-interval\":{\"Duration\":\"0s\"},\"check-interval\":{\"Duration\":\"0s\"},\"row-error-delay\":{\"Duration\":\"0s\"},\"meta-flush-interval\":{\"Duration\":\"0s\"},\"batch-query-size\":0,\"max-pending-row-size\":\"\",\"max-pending-row-count\":0},\"db-config\":{\"host\":\"mysql8-0.sources\",\"port\":3306,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":null,\"security\":null},\"server-id\":429515231,\"flavor\":\"mysql\",\"enable-gtid\":false,\"case-sensitive\":true}],\"experimental\":{\"async-checkpoint-flush\":false},\"meta-schema\":\"dm_meta\",\"online-ddl\":false,\"column-mappings\":{},\"mod-revision\":0,\"NeedExtStorage\":true}"] [cost=1] [resources="[]"] [master-id=f9c924eb-248a-414a-89ad-5c2529f90ca9]
[2022/10/27 07:39:34.470 +00:00] [INFO] [master.go:724] [CreateWorker] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [worker-type=DMDumpTask] [worker-config="{\"task-mode\":\"all\",\"shard-mode\":\"\",\"ignore-checking-items\":[],\"timezone\":\"\",\"collation_compatible\":\"loose\",\"target-database\":{\"host\":\"tidb-0.tidb\",\"port\":4000,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":{\"tidb_txn_mode\":\"optimistic\"},\"security\":null},\"shadow-table-rules\":[],\"trash-table-rules\":[],\"filters\":{},\"expression-filter\":{},\"block-allow-list\":{\"balist-01\":{\"do-tables\":[],\"do-dbs\":[\"dmjob\"],\"ignore-tables\":[],\"ignore-dbs\":[]}},\"mydumpers\":{},\"loaders\":{},\"syncers\":{},\"routes\":{},\"validators\":{},\"upstreams\":[{\"SourceID\":\"replica-02\",\"Meta\":null,\"FilterRules\":null,\"ColumnMappingRules\":null,\"RouteRules\":null,\"ExpressionFilters\":null,\"BWListName\":\"\",\"BAListName\":\"balist-01\",\"MydumperConfigName\":\"\",\"Mydumper\":{\"mydumper-path\":\"./bin/mydumper\",\"threads\":4,\"chunk-filesize\":\"64\",\"statement-size\":0,\"rows\":0,\"where\":\"\",\"skip-tz-utc\":true,\"extra-args\":\"\"},\"MydumperThread\":0,\"LoaderConfigName\":\"\",\"Loader\":{\"pool-size\":16,\"dir\":\"./dumped_data\",\"import-mode\":\"sql\",\"on-duplicate\":\"replace\"},\"LoaderThread\":0,\"SyncerConfigName\":\"\",\"Syncer\":{\"meta-file\":\"\",\"worker-count\":16,\"batch\":100,\"queue-size\":1024,\"checkpoint-flush-interval\":30,\"compact\":false,\"multiple-rows\":false,\"max-retry\":0,\"auto-fix-gtid\":false,\"enable-gtid\":false,\"disable-detect\":false,\"safe-mode\":false,\"safe-mode-duration\":\"60s\",\"enable-ansi-quotes\":false},\"SyncerThread\":0,\"ContinuousValidatorConfigName\":\"\",\"ContinuousValidator\":{\"mode\":\"none\",\"worker-count\":0,\"validate-interval\":{\"Duration\":\"0s\"},\"check-interval\":{\"Duration\":\"0s\"},\"row-error-delay\":{\"Duration\":\"0s\"},\"meta-flush-interval\":{\"Duration\":\"0s\"},\"batch-query-size\":0,\"max-pending-row-size\":\"\",\"max-pending-row-count\":0},\"db-config\":{\"host\":\"mysql8-0.sources\",\"port\":3306,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":null,\"security\":null},\"server-id\":429515231,\"flavor\":\"mysql\",\"enable-gtid\":false,\"case-sensitive\":true}],\"experimental\":{\"async-checkpoint-flush\":false},\"meta-schema\":\"dm_meta\",\"online-ddl\":false,\"column-mappings\":{},\"mod-revision\":0,\"NeedExtStorage\":true}"] [master-id=f9c924eb-248a-414a-89ad-5c2529f90ca9]
[2022/10/27 07:39:34.476 +00:00] [INFO] [worker.go:75] ["new dm worker"] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [worker_type=DMDumpTask] [worker_id=ac9f4aaf-a7a4-4975-8927-b02c9f849b05] [task_config="{\"task-mode\":\"all\",\"shard-mode\":\"\",\"ignore-checking-items\":[],\"timezone\":\"\",\"collation_compatible\":\"loose\",\"target-database\":{\"host\":\"tidb-0.tidb\",\"port\":4000,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":{\"tidb_txn_mode\":\"optimistic\"},\"security\":null},\"shadow-table-rules\":[],\"trash-table-rules\":[],\"filters\":{},\"expression-filter\":{},\"block-allow-list\":{\"balist-01\":{\"do-tables\":[],\"do-dbs\":[\"dmjob\"],\"ignore-tables\":[],\"ignore-dbs\":[]}},\"mydumpers\":{},\"loaders\":{},\"syncers\":{},\"routes\":{},\"validators\":{},\"upstreams\":[{\"SourceID\":\"replica-02\",\"Meta\":null,\"FilterRules\":null,\"ColumnMappingRules\":null,\"RouteRules\":null,\"ExpressionFilters\":null,\"BWListName\":\"\",\"BAListName\":\"balist-01\",\"MydumperConfigName\":\"\",\"Mydumper\":{\"mydumper-path\":\"./bin/mydumper\",\"threads\":4,\"chunk-filesize\":\"64\",\"statement-size\":0,\"rows\":0,\"where\":\"\",\"skip-tz-utc\":true,\"extra-args\":\"\"},\"MydumperThread\":0,\"LoaderConfigName\":\"\",\"Loader\":{\"pool-size\":16,\"dir\":\"./dumped_data\",\"import-mode\":\"sql\",\"on-duplicate\":\"replace\"},\"LoaderThread\":0,\"SyncerConfigName\":\"\",\"Syncer\":{\"meta-file\":\"\",\"worker-count\":16,\"batch\":100,\"queue-size\":1024,\"checkpoint-flush-interval\":30,\"compact\":false,\"multiple-rows\":false,\"max-retry\":0,\"auto-fix-gtid\":false,\"enable-gtid\":false,\"disable-detect\":false,\"safe-mode\":false,\"safe-mode-duration\":\"60s\",\"enable-ansi-quotes\":false},\"SyncerThread\":0,\"ContinuousValidatorConfigName\":\"\",\"ContinuousValidator\":{\"mode\":\"none\",\"worker-count\":0,\"validate-interval\":{\"Duration\":\"0s\"},\"check-interval\":{\"Duration\":\"0s\"},\"row-error-delay\":{\"Duration\":\"0s\"},\"meta-flush-interval\":{\"Duration\":\"0s\"},\"batch-query-size\":0,\"max-pending-row-size\":\"\",\"max-pending-row-count\":0},\"db-config\":{\"host\":\"mysql8-0.sources\",\"port\":3306,\"user\":\"root\",\"max-allowed-packet\":null,\"session\":null,\"security\":null},\"server-id\":429515231,\"flavor\":\"mysql\",\"enable-gtid\":false,\"case-sensitive\":true}],\"experimental\":{\"async-checkpoint-flush\":false},\"meta-schema\":\"dm_meta\",\"online-ddl\":false,\"column-mappings\":{},\"mod-revision\":0,\"NeedExtStorage\":true}"]
[2022/10/27 07:39:34.476 +00:00] [INFO] [fake_worker.go:145] ["FakeWorker: Tick"] [worker-id=bbf8d4e6-e595-4513-adb0-3c0c51dea648] [tick=139]
[2022/10/27 07:39:34.480 +00:00] [INFO] [task_runner.go:222] ["Launching task"] [id=ac9f4aaf-a7a4-4975-8927-b02c9f849b05] [runtime-task-count=4]
[2022/10/27 07:39:34.480 +00:00] [INFO] [worker_creator.go:151] ["Dispatch Worker succeeded"] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [args="{\"ProjectInfo\":{},\"WorkerID\":\"ac9f4aaf-a7a4-4975-8927-b02c9f849b05\",\"MasterID\":\"f9c924eb-248a-414a-89ad-5c2529f90ca9\",\"WorkerType\":10,\"WorkerConfig\":\"dGFzay1tb2RlID0gImFsbCIKc2hhcmQtbW9kZSA9ICIiCmlnbm9yZS1jaGVja2luZy1pdGVtcyA9IFtdCnRpbWV6b25lID0gIiIKY29sbGF0aW9uX2NvbXBhdGlibGUgPSAibG9vc2UiCnNoYWRvdy10YWJsZS1ydWxlcyA9IFtdCnRyYXNoLXRhYmxlLXJ1bGVzID0gW10KbWV0YS1zY2hlbWEgPSAiZG1fbWV0YSIKb25saW5lLWRkbCA9IGZhbHNlCm1vZC1yZXZpc2lvbiA9IDAKTmVlZEV4dFN0b3JhZ2UgPSB0cnVlCgpbdGFyZ2V0LWRhdGFiYXNlXQogIGhvc3QgPSAidGlkYi0wLnRpZGIiCiAgcG9ydCA9IDQwMDAKICB1c2VyID0gInJvb3QiCiAgcGFzc3dvcmQgPSAiIgogIFt0YXJnZXQtZGF0YWJhc2Uuc2Vzc2lvbl0KICAgIHRpZGJfdHhuX21vZGUgPSAib3B0aW1pc3RpYyIKCltmaWx0ZXJzXQoKW2V4cHJlc3Npb24tZmlsdGVyXQoKW2Jsb2NrLWFsbG93LWxpc3RdCiAgW2Jsb2NrLWFsbG93LWxpc3QuYmFsaXN0LTAxXQogICAgZG8tdGFibGVzID0gW10KICAgIGRvLWRicyA9IFsiZG1qb2IiXQogICAgaWdub3JlLXRhYmxlcyA9IFtdCiAgICBpZ25vcmUtZGJzID0gW10KCltteWR1bXBlcnNdCgpbbG9hZGVyc10KCltzeW5jZXJzXQoKW3JvdXRlc10KClt2YWxpZGF0b3JzXQoKW1t1cHN0cmVhbXNdXQogIFNvdXJjZUlEID0gInJlcGxpY2EtMDIiCiAgQldMaXN0TmFtZSA9ICIiCiAgQkFMaXN0TmFtZSA9ICJiYWxpc3QtMDEiCiAgTXlkdW1wZXJDb25maWdOYW1lID0gIiIKICBNeWR1bXBlclRocmVhZCA9IDAKICBMb2FkZXJDb25maWdOYW1lID0gIiIKICBMb2FkZXJUaHJlYWQgPSAwCiAgU3luY2VyQ29uZmlnTmFtZSA9ICIiCiAgU3luY2VyVGhyZWFkID0gMAogIENvbnRpbnVvdXNWYWxpZGF0b3JDb25maWdOYW1lID0gIiIKICBzZXJ2ZXItaWQgPSA0Mjk1MTUyMzEKICBmbGF2b3IgPSAibXlzcWwiCiAgZW5hYmxlLWd0aWQgPSBmYWxzZQogIGNhc2Utc2Vuc2l0aXZlID0gdHJ1ZQogIFt1cHN0cmVhbXMuTXlkdW1wZXJdCiAgICBteWR1bXBlci1wYXRoID0gIi4vYmluL215ZHVtcGVyIgogICAgdGhyZWFkcyA9IDQKICAgIGNodW5rLWZpbGVzaXplID0gIjY0IgogICAgc3RhdGVtZW50LXNpemUgPSAwCiAgICByb3dzID0gMAogICAgd2hlcmUgPSAiIgogICAgc2tpcC10ei11dGMgPSB0cnVlCiAgICBleHRyYS1hcmdzID0gIiIKICBbdXBzdHJlYW1zLkxvYWRlcl0KICAgIHBvb2wtc2l6ZSA9IDE2CiAgICBkaXIgPSAiLi9kdW1wZWRfZGF0YSIKICAgIGltcG9ydC1tb2RlID0gInNxbCIKICAgIG9uLWR1cGxpY2F0ZSA9ICJyZXBsYWNlIgogIFt1cHN0cmVhbXMuU3luY2VyXQogICAgbWV0YS1maWxlID0gIiIKICAgIHdvcmtlci1jb3VudCA9IDE2CiAgICBiYXRjaCA9IDEwMAogICAgcXVldWUtc2l6ZSA9IDEwMjQKICAgIGNoZWNrcG9pbnQtZmx1c2gtaW50ZXJ2YWwgPSAzMAogICAgY29tcGFjdCA9IGZhbHNlCiAgICBtdWx0aXBsZS1yb3dzID0gZmFsc2UKICAgIG1heC1yZXRyeSA9IDAKICAgIGF1dG8tZml4LWd0aWQgPSBmYWxzZQogICAgZW5hYmxlLWd0aWQgPSBmYWxzZQogICAgZGlzYWJsZS1kZXRlY3QgPSBmYWxzZQogICAgc2FmZS1tb2RlID0gZmFsc2UKICAgIHNhZmUtbW9kZS1kdXJhdGlvbiA9ICI2MHMiCiAgICBlbmFibGUtYW5zaS1xdW90ZXMgPSBmYWxzZQogIFt1cHN0cmVhbXMuQ29udGludW91c1ZhbGlkYXRvcl0KICAgIG1vZGUgPSAibm9uZSIKICAgIHdvcmtlci1jb3VudCA9IDAKICAgIHZhbGlkYXRlLWludGVydmFsID0gIjBzIgogICAgY2hlY2staW50ZXJ2YWwgPSAiMHMiCiAgICByb3ctZXJyb3ItZGVsYXkgPSAiMHMiCiAgICBtZXRhLWZsdXNoLWludGVydmFsID0gIjBzIgogICAgYmF0Y2gtcXVlcnktc2l6ZSA9IDAKICAgIG1heC1wZW5kaW5nLXJvdy1zaXplID0gIiIKICAgIG1heC1wZW5kaW5nLXJvdy1jb3VudCA9IDAKICAgIHN0YXJ0LXRpbWUgPSAiIgogIFt1cHN0cmVhbXMuZGItY29uZmlnXQogICAgaG9zdCA9ICJteXNxbDgtMC5zb3VyY2VzIgogICAgcG9ydCA9IDMzMDYKICAgIHVzZXIgPSAicm9vdCIKICAgIHBhc3N3b3JkID0gIiIKCltleHBlcmltZW50YWxdCiAgYXN5bmMtY2hlY2twb2ludC1mbHVzaCA9IGZhbHNlCgpbY29sdW1uLW1hcHBpbmdzXQo=\",\"WorkerEpoch\":23}"]
[2022/10/27 07:39:34.481 +00:00] [INFO] [worker.go:133] ["initializing the dm worker"] [job_id=f9c924eb-248a-414a-89ad-5c2529f90ca9] [worker_id=ac9f4aaf-a7a4-4975-8927-b02c9f849b05] [task-id=replica-02]
[2022/10/27 07:39:34.493 +00:00] [INFO] [file_manager.go:198] ["Removing resource"] [ident="{\"Executor\":\"chaos-executor-2-8b478d09\",\"WorkerID\":\"4a1992be-087e-4ce1-9593-8c6d5e87a7ab\",\"Name\":\"66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032\"}"]
[2022/10/27 07:39:34.494 +00:00] [INFO] [s3.go:384] ["succeed to get bucket region from s3"] ["bucket region"=us-east-1]
[2022/10/27 07:39:34.498 +00:00] [INFO] [file_manager.go:272] ["Removing resources"] [scope="{\"Executor\":\"chaos-executor-2-8b478d09\",\"WorkerID\":\"4a1992be-087e-4ce1-9593-8c6d5e87a7ab\"}"] [toRemoveFiles="[66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/.keep,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob-schema-create.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb0-schema.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb0.0000000000000.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb1-schema.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb1.0000000000000.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb2-schema.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb2.0000000000000.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb3-schema.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb3.0000000000000.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb4-schema.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/dmjob.tb4.0000000000000.sql,66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032/metadata]"]
[2022/10/27 07:39:34.518 +00:00] [PANIC] [file_manager.go:242] ["inconsistent executor ID of s3 file"] [resource="{\"Executor\":\"chaos-executor-2-8b478d09\",\"WorkerID\":\"4a1992be-087e-4ce1-9593-8c6d5e87a7ab\",\"Name\":\"66396339323465622d323438612d343134612d383961642d3563323532396639306361392d7265706c6963612d3032\"}"] [creator="\"chaos-executor-2-8b478d09\""] [currentExecutor=chaos-executor-0-da471afe] [stack="github.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3.(*FileManager).validateExecutor\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3/file_manager.go:242\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3.(*FileManager).CreateResource\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3/file_manager.go:95\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/broker.(*DefaultBroker).getPersistResource\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/broker/broker.go:352\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/broker.(*DefaultBroker).OpenStorage\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/broker/broker.go:174\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseWorker).OpenStorage\n\tgithub.com/pingcap/tiflow/engine/framework/worker.go:520\ngithub.com/pingcap/tiflow/engine/executor/dm.(*dmWorker).setupStorage\n\tgithub.com/pingcap/tiflow/engine/executor/dm/worker.go:203\ngithub.com/pingcap/tiflow/engine/executor/dm.(*dmWorker).InitImpl\n\tgithub.com/pingcap/tiflow/engine/executor/dm/worker.go:140\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseWorker).Init\n\tgithub.com/pingcap/tiflow/engine/framework/worker.go:262\ngithub.com/pingcap/tiflow/engine/framework/internal/eventloop.(*Runner[...]).doRun\n\tgithub.com/pingcap/tiflow/engine/framework/internal/eventloop/runner.go:84\ngithub.com/pingcap/tiflow/engine/framework/internal/eventloop.(*Runner[...]).Run\n\tgithub.com/pingcap/tiflow/engine/framework/internal/eventloop/runner.go:59\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).launchTask.func1\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:226"]
[2022/10/27 07:39:34.518 +00:00] [ERROR] [task_runner.go:208] ["Task panicked"] [id=ac9f4aaf-a7a4-4975-8927-b02c9f849b05] [error="panic: inconsistent executor ID of s3 file"] [errorVerbose="panic: inconsistent executor ID of s3 file\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).launchTask.func1.1\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:207\nruntime.gopanic\n\truntime/panic.go:884\ngo.uber.org/zap/zapcore.CheckWriteAction.OnWrite\n\tgo.uber.org/zap@v1.23.0/zapcore/entry.go:198\ngo.uber.org/zap/zapcore.(*CheckedEntry).Write\n\tgo.uber.org/zap@v1.23.0/zapcore/entry.go:264\ngo.uber.org/zap.(*Logger).Panic\n\tgo.uber.org/zap@v1.23.0/logger.go:251\ngithub.com/pingcap/log.Panic\n\tgithub.com/pingcap/log@v1.1.0/global.go:54\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3.(*FileManager).validateExecutor\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3/file_manager.go:242\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3.(*FileManager).CreateResource\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/internal/s3/file_manager.go:95\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/broker.(*DefaultBroker).getPersistResource\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/broker/broker.go:352\ngithub.com/pingcap/tiflow/engine/pkg/externalresource/broker.(*DefaultBroker).OpenStorage\n\tgithub.com/pingcap/tiflow/engine/pkg/externalresource/broker/broker.go:174\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseWorker).OpenStorage\n\tgithub.com/pingcap/tiflow/engine/framework/worker.go:520\ngithub.com/pingcap/tiflow/engine/executor/dm.(*dmWorker).setupStorage\n\tgithub.com/pingcap/tiflow/engine/executor/dm/worker.go:203\ngithub.com/pingcap/tiflow/engine/executor/dm.(*dmWorker).InitImpl\n\tgithub.com/pingcap/tiflow/engine/executor/dm/worker.go:140\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseWorker).Init\n\tgithub.com/pingcap/tiflow/engine/framework/worker.go:262\ngithub.com/pingcap/tiflow/engine/framework/internal/eventloop.(*Runner[...]).doRun\n\tgithub.com/pingcap/tiflow/engine/framework/internal/eventloop/runner.go:84\ngithub.com/pingcap/tiflow/engine/framework/internal/eventloop.(*Runner[...]).Run\n\tgithub.com/pingcap/tiflow/engine/framework/internal/eventloop/runner.go:59\ngithub.com/pingcap/tiflow/engine/executor/worker.(*TaskRunner).launchTask.func1\n\tgithub.com/pingcap/tiflow/engine/executor/worker/task_runner.go:226\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
[2022/10/27 07:39:34.518 +00:00] [INFO] [task_runner.go:210] ["Task Closed"] [id=ac9f4aaf-a7a4-4975-8927-b02c9f849b05] [] [runtime-task-count=4]
```
### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

```console
(paste Dataflow Engine version here)
```

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #7486

### What is changed and how it works?
1. add cleanPersistResource method for filemananger

#### Tests 

 - Unit test
 - Integration test
 
### Release note 

```release-note
`None`
```

### CODE REVIEW

File: engine/pkg/externalresource/broker/broker.go [hunk 1]
```diff
@@ -170,8 +170,10 @@ func (b *DefaultBroker) OpenStorage(
 	var desc internal.ResourceDescriptor
 	if !exists {
 		desc, err = b.createResource(ctx, fm, projectInfo, workerID, resName)
+	} else if !options.cleanBeforeOpen {
+		desc, err = b.getPersistResource(ctx, fm, record, resName)
 	} else {
-		desc, err = b.getPersistResource(ctx, fm, record, resName, options)
+		desc, err = b.cleanPersistResource(ctx, fm, record, resName)
```
Comment (amyangfei) [inline engine/pkg/externalresource/broker/broker.go:176]: Seems `cleanPersistResource` does the following work. Maybe we can rename this function and make it easier to understand.
- Clean existing files if they exist
- Re-create resource if possible.

**Review by amyangfei — APPROVED** [review]

File: engine/pkg/externalresource/internal/file_manager.go [hunk 2]
```diff
@@ -77,6 +77,9 @@ type FileManager interface {
 	// GetPersistedResource returns the descriptor of an already persisted resource.
 	GetPersistedResource(ctx context.Context, ident ResourceIdent) (ResourceDescriptor, error)
 
+	// CleanOrRecreatePersistedResource cleans the persisted resource.
+	CleanOrRecreatePersistedResource(ctx context.Context, ident ResourceIdent) (ResourceDescriptor, error)
```
Comment (gozssky) [inline engine/pkg/externalresource/internal/file_manager.go:81]: `CleanOrRecreatePersistedResource` sounds like it's doing either cleaning or recreating, but it actually does both.

Maybe just `RecreatePersistedResource` is enough.

Comment (CharlesCheung96) [reply engine/pkg/externalresource/internal/file_manager.go:81]: For local filemanager, it simply removes the resource and recreates it. 
For s3 filemanager, it either [cleans files](https://github.com/pingcap/tiflow/pull/7491/files#diff-1c2eff62ba26b13484fc11619e06322ed96dd751247d7fd0523946030aacbe37R157) or [rebuilds resources](https://github.com/pingcap/tiflow/pull/7491/files#diff-1c2eff62ba26b13484fc11619e06322ed96dd751247d7fd0523946030aacbe37R141).

File: engine/pkg/externalresource/broker/broker.go [hunk 3]
```diff
@@ -340,23 +341,27 @@ func (b *DefaultBroker) getPersistResource(
...
-		// LocalFileManager may return ErrResourceDoesNotExist, which can be
-		// ignored because the resource no longer exists.
-		if err != nil && !derrors.ErrResourceDoesNotExist.Equal(err) {
-			return nil, err
-		}
-		desc, err = fm.CreateResource(ctx, ident)
-		if err != nil {
-			return nil, err
-		}
-		return desc, nil
```
Comment (gozssky) [inline engine/pkg/externalresource/broker/broker.go:356]: What's wrong with the old logic? Remove and Create seems to be the most straightforward approach to clean the resource.

File: engine/pkg/externalresource/internal/file_manager.go (hunk 2 above)
Comment (gozssky) [reply engine/pkg/externalresource/internal/file_manager.go:81]: Please update the comment on this method.

File: engine/pkg/externalresource/broker/broker.go (hunk 3 above)
Comment (CharlesCheung96) [reply engine/pkg/externalresource/broker/broker.go:356]: The core problem, in terms of interface semantics, is that `CreateResource ` can only be used to create resources that belong to the current executor.
ref: https://github.com/pingcap/tiflow/blob/master/engine/pkg/externalresource/internal/s3/file_manager.go#L95

File: engine/pkg/externalresource/internal/s3/file_path_util.go [hunk 4]
```diff
@@ -26,12 +26,17 @@ func getPathPredAlwaysTrue() pathPredFunc {
 	}
 }
 
-func getPathPredByName(target string) pathPredFunc {
+func getPathPredByName(target string, keepPlaceholder bool) pathPredFunc {
```
Comment (gozssky) [inline engine/pkg/externalresource/internal/s3/file_path_util.go:29]: Nit: The meaning of `getPathPredByName` is not clear when it leaves the scope of `removeFilesIf`.

**Review by gozssky — APPROVED** [review]

**Comment by amyangfei** [comment]
wait chaos test: https://github.com/pingcap/tiflow/actions/runs/3390924082

File: engine/pkg/externalresource/internal/s3/file_path_util.go (hunk 4 above)
Comment (CharlesCheung96) [reply engine/pkg/externalresource/internal/s3/file_path_util.go:29]: Currently `getPathPredByName` is only used in the `removeFilesIf ` function, perhaps we could rename it to a more appropriate name when we extend the application. In addition, do you have a better name to recommend?

**Pull request merged**